interview-questions.md

```diff
@@ -7,52 +7,52 @@ Then, research the question to expand on your answer. Even if you feel you have
 
 1. What are props in React?
 
-  Your answer:
+  Your answer: Props is short for properties. this is used as from parent to child components. meaning to pass from one component to another. Props are both used with and without destructuring. Props is for
+  
+  Researched answer: Javascript library for building user interface free and open source. Use is for state management and rendering the state of the DOM structure in mobile applications or even server rendering. React is in entities called components that are React code. this is declarative programming. Class based components, virtual DOM, lifecycle methods, JSX, React hooks are noteworthy features of React.
 
-  Researched answer:
 
 
+2. What is a DOM event? 
+  
+  Your answer: DOM stands for document object model DOM events are used in conjuntion with functions. Event occurs then the function happens.
+  for example "click" this happens when the user clicks on a certain element.
 
-2. What is a DOM event?
-
-  Your answer:
-
-  Researched answer:
+  Researched answer: DOM events are generated by element nodes and HTML events like click, dblclick, mousedown, mouseup are examples of DOM events. Elements are inside the node of DOM trees and are triggered by user interaction to the browser.
 
 
 
 3. What is object-oriented programming? How is it different than functional programming?
 
-  Your answer:
+  Your answer: The difference between object-oriented programming and functional programming is that in functional programming the focus is on logic and functions. Object-oriented programming is more about objects data modeling. once labeled with a class uses are for data manipulation.
 
-  Researched answer:
+  Researched answer: Object oriented programming based on objects. These object attributes have code then formulate a method for the procedure. "This" refers to the current oject. Function in comparison uses functions. the logic of highorder function is an example of usage.
 
 
 
 4. What is the difference between a Float and an Integer in Ruby?
 
-  Your answer:
-
-  Researched answer:
+  Your answer: A float is a decimal and an integer is no decimal point.
 
+  Researched answer: Integer class is a class that holds whole numbers and is a subclass of numeric class. Methods that are defined under integer class would be integer?, next, times, upto. Any number that is not a decimal point will be treated as an integer. Floats contain decimal points in Ruby and is also a subclass in numeric class.
 
 
 5. Ruby has an implicit return. What does that mean?
 
-  Your answer:
+  Your answer: The return is implied without the function.
```

Comment:
Make sure to extend these answers further.  This is meant to mimic how you would respond in an interview, and one-line answers are not very conversational and could come off as uninterested.